### ISSUE TITLE
Error on loading remote using a SDK with multiple entry points

### ISSUE THREAD

**Issue by lbroschart** [module-federation/vite#97]
First of all thank you for this plugin!

In our company we try to set up a federated frontend while using our React Library as shared dependency. Unfortunately we have problems with several context providers offered by our SDK. These providers are mainly Material UI based Context like the ThemingProvider or LocalizationProvider (Date-Picker).

This leads to the App either not rendering because:
```
Uncaught Error: MUI X: Can not find the date and time pickers localization context.
It looks like you forgot to wrap your component in LocalizationProvider.
````
or the Theme not be applied.

After much time of debugging and trying I was able to pin the problem on the way our SDK exports its components.
We are using multiple entry points which seem to cause the context not to be shared correctly as singleton.

* Context Provider Wrapper exported from main:
`import { SDKWrapper } from 'example-sdk'`
* Components using context exported from different entry point: 
`import { Button, CustomDatePicker } from 'example-sdk/components'`

I created a small Repo for reproduction of this problem: 
https://github.com/lbroschart/vite-federation-sdk-example (See README for more information)

TL;DR: If everything is imported from the SDK main index it works, if imported from different entries the mentioned errors occur.

Maybe I did missed something on how to correctly configure the shared dependencies? Is there anything we can do in this case other than refactor our SDK to export everything on the same entry point (which would be the last thing we like to do, since our real SDK has even more entries then the example)?

Thank you in advance for any help on this issue!

**Comment by zhangHongEn** [comment]
This is a bit complicated, so let me explain.

**Comment by zhangHongEn** [comment]
In conclusion, to ensure that all modules in shared are singletons, you can set dependencies like react, react-dom, example-sdk, and @mui/... as both external and shared. This should generally work fine, but you may encounter other issues.

The simplest approach for now is:

In the host’s vite.config:
`shared: ['react', 'react-dom', 'example-sdk', 'example-sdk/']`
In the host’s src:
`
// The host only needs to reference these dependencies to register them into `sharedScopes`. Then, the dependencies retrieved in the remote will be guaranteed to be singletons.
() => import('example-sdk');
() => import('example-sdk/components');
`

In the remote’s vite.config:
`shared: ['react', 'react-dom', 'example-sdk', 'example-sdk/']`

**Comment by lbroschart** [comment]
It did work, we now have no errors in our current federation project. 🥳 
Thank you very much!
